### PR TITLE
Refactor and add collectl

### PR DESCRIPTION
**Pull request by dustinblack**
## Changes introduced with this PR

This is largely a refactor to meet current standards from the plugin template. Additionally, this changes the `pmlogger.conf` file to be generated instead of included, and it adds the missing collectl service.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: arcaflow_plugin_pcp/pcp_schema.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            "pmlogger for data collection"
+        ),
+    ] = None
+
+
+@dataclass
+class PerfOutput:
+    pcp_output: typing.Annotated[
+        typing.Dict[str, typing.Any],
+        schema.name("PCP output dictionary"),
```
Comment (jaredoconnell) [inline arcaflow_plugin_pcp/pcp_schema.py:29]: Calling it a directory could cause a little confusion, because it's not a file directory, right?

**Review by jaredoconnell — APPROVED** [review]
Looks good to me. I left one comment for a small concern.

File: arcaflow_plugin_pcp/pcp_schema.py (hunk 1 above)
Comment (dustinblack) [reply arcaflow_plugin_pcp/pcp_schema.py:29]: I'm not following. What's being called a directory?

Comment (jaredoconnell) [reply arcaflow_plugin_pcp/pcp_schema.py:29]: Ah, I misread it. Nevermind.

**Review by Harshith-umesh — APPROVED** [review]
LGTM!

**Pull request merged**